src/Application/Controller/Project/ProjectController.php

```diff
@@ -212,6 +212,26 @@ public function projectLike(Request $request, string $id): Response
     ]);
   }
 
+  /**
+   * @throws NoResultException
+   */
+  #[Route(path: '/project/steal/{id}', name: 'project_steal', methods: ['GET'])]
```

Comment:
At Catroweb we try to write our code using the API approach, to decouple the frontend from the backend. We have multiple different frontends (ours on the share, and also the apps)
Usually, your request (steal) would use our API (https://github.com/Catrobat/Catroweb-API // https://developer.catrobat.org/Catroweb-API/) Of course that is an oerhead for the training example.

Instead of using a GET request, consider using POST which is used to send data to a server to create/update a resource. Feel free to take a look at all available methods: https://www.w3schools.com/tags/ref_httpmethods.asp

That said it would be nice, to return a status code rather than a redirect in the response, and handle the response using the frontend logic.